tests/unit/development/metadata_test.py

```diff
@@ -91,7 +92,31 @@ class Table(Component):
     _namespace = 'TableComponents'
     _type = 'Table'
     @_explicitize_args
-    def __init__(self, children=None, optionalArray=Component.UNDEFINED, optionalBool=Component.UNDEFINED, optionalFunc=Component.UNDEFINED, optionalNumber=Component.UNDEFINED, optionalObject=Component.UNDEFINED, optionalString=Component.UNDEFINED, optionalSymbol=Component.UNDEFINED, optionalNode=Component.UNDEFINED, optionalElement=Component.UNDEFINED, optionalMessage=Component.UNDEFINED, optionalEnum=Component.UNDEFINED, optionalUnion=Component.UNDEFINED, optionalArrayOf=Component.UNDEFINED, optionalObjectOf=Component.UNDEFINED, optionalObjectWithExactAndNestedDescription=Component.UNDEFINED, optionalObjectWithShapeAndNestedDescription=Component.UNDEFINED, optionalAny=Component.UNDEFINED, customProp=Component.UNDEFINED, customArrayProp=Component.UNDEFINED, id=Component.UNDEFINED, **kwargs):
+    def __init__(
+        self,
+        children: typing.Union[str, int, float, Component, typing.List[typing.Union[str, int, float, Component]]] = None,
+        optionalArray: typing.List = Component.UNDEFINED,
+        optionalBool: bool = Component.UNDEFINED,
+        optionalFunc: typing.Any = Component.UNDEFINED,
+        optionalNumber: typing.Union[float, int] = Component.UNDEFINED,
```

Comment:
@alexcjohnson @T4rk1n just wondering if people can pass in Decimals in here as well?

Reply:
If [`to_json_plotly`](https://github.com/plotly/dash/blob/38cd3390ba59cc7fb2a5d1195047b704766e9079/dash/_utils.py#L18-L22) can serialize it then yes. It's pretty flexible, so probably `Decimal` is fine, but I haven't tested it.

Reply:
`Decimal` can serialize, I think we can put `numbers.Number` to get all the number types.

Reply:
`numbers.Number` is too much, it includes `complex` - how about `numbers.Real`?

Reply:
https://docs.python.org/3/library/numbers.html#the-numeric-tower

Number is the base class, then they extend the last one, to get the `Integral` you need all the other 3 implemented. Tested with numpy and Number is what get the most support and our `to_json` supports them, I dont think custom implementation of those works.

Reply:
hmph ok, `numbers.Real` doesn't catch `Decimal` for some reason, but `numbers.Number` does. So OK, we can leave it as you have it (`int, float, numbers.Number`) - better to allow some invalid items through than to prevent some valid items.